amazon/ion/ioncmodule.c

```diff
@@ -1503,43 +1527,108 @@ void ionc_read_iter_dealloc(PyObject *self) {
     PyObject_Del(self);
 }
 
+iERR ionc_lazy_read_all(hREADER hreader, PyObject* container, BOOL in_struct, BOOL emit_bare_values, char* buffer) {
+    iENTER;
+    ION_TYPE t;
+    for (;;) {
+        IONCHECK(ion_reader_next(hreader, &t));
+        if (t == tid_EOF) {
+            assert(t == tid_EOF && "next() at end");
+            break;
+        }
+        PyObject* rtn;
+        PyObject* py_cached_bytes;
+
+        // Calculate the start position and the length of the value that writer is stand on so we can cache this bytes
+        // Start position
+        POSITION p_offset = 0;
+        // Value length
+        SIZE  p_length = 0;
+        ion_reader_get_value_offset(hreader, &p_offset);
+        ion_reader_get_value_length(hreader, &p_length);
```

Comment:
Nice, I forgot ion-c provided this functionality. It might not be well-tested, however... That is worth looking into. We want to make sure it's accurate in all cases.